app/js/lib/server/settings.js

```diff
@@ -0,0 +1,236 @@
+'use strict';
+
+import EventDispatcher from '../common/event-dispatcher';
+
+// Prefix all entries to avoid collisions.
+const PREFIX = 'calendar-';
+
+const ORIGIN = 'https://calendar.knilxof.org';
+
+/**
+ * API version to use (currently not configurable).
+ * @type {number}
+ * @const
+ */
+const API_VERSION = 1;
+
+/**
+ * Regex to match upper case literals.
+ * @type {RegExp}
+ * @const
+ */
+const UPPER_CASE_REGEX = /([A-Z])/g;
+
+const p = Object.freeze({
+  values: Symbol('values'),
+  storage: Symbol('storage'),
+
+  // Private methods.
+  updateSetting: Symbol('updateSetting'),
+  stringToSettingTypedValue: Symbol('stringToSettingTypedValue'),
+  getDefaultSettingValue: Symbol('getDefaultSettingValue'),
+  onStorage: Symbol('onStorage'),
+});
+
+// Definition of all available settings and their default values (if needed).
+const settings = Object.freeze({
+  // String settings.
+  SESSION: Object.freeze({ key: 'session' }),
```

Comment:
I see your point, but we don't always have a default value to extract the type from.
But if you think this change will make things easier, you're welcome to work on it or to open an issue to not forget about it.